### PR TITLE
Scripts: Fix style.css handling in the build and start commands

### LINKED ISSUE #22776: importing style.(c/sc)ss with @wordpress/scripts 10.0 causes JavaScript to stop running?
**Describe the bug**
Native support for CSS/style loaders was just added to @wordpress/scripts in 10.0 via #21730. This is mad awesome and I've been keeping an eye on @gziolo 's + @fabiankaegy 's progress for a while, but I'm running into an issue where the resulting build/index.js' code doesn't run when the entry point (or anything it imports `import`s a `./style.css`). For example, `registerBlockType` can be right after the `import ./style.scss` and the block will not show up in the block inserter. Importing other CSS files, no matter the name, doesn't appear to disrupt the code.

I think it may have to do with the way Webpack uses `splitChunks`/`cacheGroups` on `style` ([webpack.config.js](https://github.com/WordPress/gutenberg/blob/master/packages/scripts/config/webpack.config.js)). Whenever a `style.css` is imported, it adds some "deferredModules` stuff to the built JS and it maybe it just isn't getting to it?

**To reproduce**
I tried this in a fresh plugin closely replicating what was done on #21730's _How has this been tested?_:
1. `cd` to `wp-content/plugins` and create a new block plugin with `npm init @wordpress/block yo-whats-good`
2. `cd yo-whats-good` and `npm run start` (no issues here--watch task seems to run fine)
3. Activate _Yo Whats Good_ and _Gutenberg_ plugins and open up `page` post in block editor
4. Add a new file `yo-whats-good/src/editor.scss` with some styles for the editor.
5. Add a new file `yo-whats-good/src/style.scss` with some styles for both the front-end and the editor.
6. Open `yo-whats-good/src/index.js` and add CSS imports the beginning:
```
import './editor.scss';
import './style.scss';
``` 
7. Save and let the `start` script watch process run
8. Refresh page editor and try to insert the custom block. It should not appear in inserter.
9. Comment-out `import './style.scss;`, save, and refresh the page editor. The registered block should appear.

**Expected behavior**
The creation of two new CSS files in the `/build` directory: `index.css` and `style.css` (this happens), as well as an `index.js` (also happens) with block registration, a console log, or anything else that runs when enqueued (block doesn't end up registering, console logs don't happen).

**Editor version (please complete the following information):**
- WordPress version: 5.4.1
- Gutenberg plugin version 8.2.1

**Desktop (please complete the following information):**
 - OS: Windows 10.0 Build 18363
 - Browser: Chrome 83.0.4103.61, Firefox Developer Edition

**Additional context**
Pull request that added CSS support to @wordpress/scripts: https://github.com/WordPress/gutenberg/pull/21730

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Fixes #22776.

Major props to @ocean9, @ryelle and @nerrad for their help and prior art I could use for the proposed fix 🙇 

It's late here so I will add more details another day :)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [n/a] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: .wp-env.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
 	"core": "WordPress/WordPress",
-	"plugins": [ "." ],
+	"plugins": [
+		"."
+	],
```
Comment (ocean90) [inline .wp-env.json:5]: Seems unrelated?

File: packages/scripts/README.md [hunk 2]
```diff
@@ -528,7 +528,7 @@ import './style.css';
 When you run the build using the default command `wp-scripts build` (also applies to `start`) in addition to the JavaScript file `index.js` generated in the `build` folder, you should see two more files:
 
 1. `index.css` – all imported CSS files are bundled into one chunk named after the entry point, which defaults to `index.js`, and thus the file created becomes `index.css`. This is for styles used only in the editor.
-2. `style.css` – imported `style.css` file(s) (applies to SASS and SCSS extensions) get bundled into one `style.css` file that is meant to be used both on the front-end and in the editor.
+2. `style-index.css` – imported `*style.css` file(s) (applies to SASS and SCSS extensions) get bundled into one `style-index.css` file that is meant to be used both on the front-end and in the editor.
```
Comment (ocean90) [inline packages/scripts/README.md:531]: Should we actually limit it to only `style.css` files?

**Review by ocean90 — APPROVED** [review]
This is looking good and it's working for me! Also tested with a custom webpack config with different entry points.

File: .wp-env.json (hunk 1 above)
Comment (gziolo) [reply .wp-env.json:5]: Yes, was testing it with `wp-env` and in retrospect, I should use `.wp-env.override.json` 😅

File: packages/scripts/README.md (hunk 2 above)
Comment (gziolo) [reply packages/scripts/README.md:531]: I guess it's fine to limit it. It's a breaking change regardless that I still need to include in the CHANGELOG file. Previously, we would bundle all `style` files into one even when there were multiple entry points. I found a way to resolve it.

File: .wp-env.json (hunk 1 above)
Comment (gziolo) [reply .wp-env.json:5]: It's VSC on save refactor, I had to fix it with the terminal :)

File: packages/scripts/README.md (hunk 2 above)
Comment (gziolo) [reply packages/scripts/README.md:531]: Well, it's an interesting case because it really never worked as expected 🤣 

We can classify as a bug fix and adjust however we want. I will test more strict regexp on Monday.

Comment (ocean90) [reply packages/scripts/README.md:531]: > Previously, we would bundle all `style` files into one even when there were multiple entry points.

Right, now we only have one per entry point which is good IMO. My comment was more about that also files like `foostyle.css` are handled by this rule which I think we should prevent.

File: packages/scripts/config/webpack.config.js [hunk 3]
```diff
@@ -64,11 +64,11 @@ const config = {
 			mode === 'production' && ! process.env.WP_BUNDLE_ANALYZER,
 		splitChunks: {
 			cacheGroups: {
-				styles: {
-					name: 'style',
+				style: {
 					test: /style\.(sc|sa|c)ss$/,
```
Comment (ocean90) [inline packages/scripts/config/webpack.config.js:68]:
```suggestion
					test: /[\\/]style\.(sc|sa|c)ss$/,
```

This should be enough to limit it to only `style.*` files.

Comment (gziolo) [reply packages/scripts/config/webpack.config.js]: Let’s test it 😃
